BaseStation/VideoStreamer/MainWindow.xaml.cs

```diff
@@ -1,29 +1,231 @@
+        private void initPorts()
+        {
+            String line;
+            int x;
+            StreamReader file = new StreamReader(portFileLocation);
+            while ((line = file.ReadLine()) != null)
+            {
+                if (Int32.TryParse(line, out x))
+                {
+                    ports.Add(x);
+                }
+            }
+            file.Close();
+        }
+
+        // initializes the processes used to launch the stream and file recorder
+        private void initProcesses(ref Process[] pro)
+        {
+            pro = new Process[ports.Count];
+            for (int i = 0; i < ports.Count; i++)
+            {
+                pro[i] = new Process();
+                pro[i].StartInfo.RedirectStandardInput = false;
+                pro[i].StartInfo.UseShellExecute = true;
+            }
+        }
+
+        // handles the time increase for the timer
+        private void dispatcherTimer_Tick(object sender, EventArgs e)
+        {
+            duration = duration.AddSeconds(1);
+            time.Content = "Recording Duration: " + duration.ToString("HH:mm:ss");
+        }
+
+        // initiates the recording of the stream
+        private void ButtonRecord(object sender, RoutedEventArgs e)
+        {
+                initProcesses(ref recorders);
+                for (int i = 0; i < ports.Count; i++)
+                {
+                    recordStream(ports.ElementAt(i), recorders[i], i);
+                }
+                duration = new DateTime(0);
+                time.Content = "Recording Duration: " + duration.ToString("HH:mm:ss");
+                recordingTime.Start();
+        }
+
+        // plays the video streams in their own windows
+        private void ButtonLaunch(object sender, RoutedEventArgs e)
+        {
+            if (playing)
+            {
+                playing = false;
+                btnLaunch.Content = "Launch";
+                closeAll(players);
+            }
+            else
+            {
+                playing = true;
+                btnLaunch.Content = "Stahp";
+                initProcesses(ref players);
+                for (int i = 0; i < ports.Count; i++)
+                {
+                    startStream(ports.ElementAt(i), players[i]);
+                }
+            }
+        }
+
+        // closes all windows in the array of processes given
+        private void closeAll(Process[] cmd)
+        {
+            for (int i = 0; i < cmd.Length; i++)
+            {
+                cmd[i].CloseMainWindow();
+            }
+        }
+
+        // initiates a stream with a given port
+        private void startStream(int port, Process cmd)
+        {
+            cmd.StartInfo.FileName = "cmd.exe";
+            cmd.StartInfo.RedirectStandardInput = true;
+            cmd.StartInfo.UseShellExecute = false;
+            cmd.Start();
+            cmd.StandardInput.WriteLine(@"set PATH=%PATH%;C:\gstreamer\1.0\x86_64\bin");
```

Comment:
It would be great to check with File.exists to see if the executable is installed correctly before trying to run it. If it's not installed, you can show an error message telling the user to install GStreamer to a specific location (and also tell the location it's looking for the executable in.)

You can test this be renaming the executable to something like "gst-launch-1.0.exe__" so it has a different file name.